### ISSUE TITLE
a11y: incorrect aria toggle state

### ISSUE THREAD

**Issue by blackfalcon** [AlaskaAirlines/auro-hyperlink#332]
### What browser client(s) are you seeing the issue on?

- [x] Firefox
- [x] Chrome
- [x] Safari
- [x] Microsoft Edge

### What devices platform(s) are you seeing the issue on?

- [x] Desktop
- [x] Mobile

### Steps to reproduce

See code:

https://github.com/AlaskaAirlines/auro-hyperlink/blob/824be32fd59c823e6d0c695bb1668ba80b590b39/src/auro-hyperlink.js#L129-L131

This component is using `aria-pressed` incorrectly, and this should not be part of the baseline HTML structure. 

The `aria-pressed` attribute should be removed. 

**User impact**
In its current state, using `role="button"`, screen readers announce this as a `toggle button` feature. But that is confusing to an AT user because what it toggles is not clear, and the toggle state is dependent on a key-pressed state, not a toggle state. 

Additionally, it was noticed that the `spacebar` is not correctly implemented for `role="button"`, creating an additional keyboard accessibility issue.

### Expected behavior

The role of `aria-pressed` should only be used if this 'button' is being used to toggle a UI. See this ARIA-WAI example https://www.w3.org/WAI/ARIA/apg/patterns/button/examples/button/ (see the MUTE button demo).

There are other scenarios where things like `aria-expanded` and `aria-controls` would be applicable, but again, these should not be part of the native HTML template, but things that can be added via author controls. 

### Which version of auro-hyperlink do you have installed?

@latest

### Reproducible on the Auro doc site

- [x] I am able to reproduce this issue on the [Auro doc site](https://auro.alaskaair.com).

**Comment by jason-capsule42** [comment]
`spacebar` behavior identified in the issue description will be addressed separately by this ticket: https://github.com/AlaskaAirlines/auro-hyperlink/issues/335
